fedora/infra.py

```diff
@@ -150,31 +197,119 @@ def format_title(outage):
-                        f"{outage['enddate'] if outage['enddate'] else 'Unknown'}{NL}"
-                    )
+            message += f"**no planned or ongoing outages on Fedora Infrastructure.**{NL}"
+            await evt.respond(message)
+            return
+
+        message += (
+            f"**{len(ongoing)} ongoing** and **{len(planned)} planned** "
+            f"outages on Fedora Infrastructure.{NL}"
+        )
+
+        if ongoing:
+            message += f"##### Ongoing{NL}"
+            for outage in ongoing:
+                start_time = self.date_validation(outage.get("startdate"))
+                end_time = (
+                    self.date_validation(outage.get("enddate"))
+                    if outage.get("enddate") is not None
+                    else None
+                )
+
+                now = datetime.datetime.now(start_time.tzinfo)
+                delta_time = now - start_time
+
+                relative_start_time = self.get_relative_time(delta_time)
+                relative_end_time = (
+                    self.get_relative_time(end_time - now) if end_time is not None else None
+                )
+
+                message += f" * {format_title(outage)}{NL}"
+                message += f"   Started : {relative_start_time}{NL}{start_time}"
+
+                message += (
+                    f"   Estimated to end: "
+                    f"{end_time if end_time else 'Unknown'}{NL}"
+                    f"{relative_end_time  if relative_end_time else 'Unknown'}{NL}"
+                )
+
             if planned:
-                message = message + f"##### Planned{NL}"
+                message += f"##### Planned{NL}"
                 for outage in planned:
-                    message = message + f" * {format_title(outage)}{NL}"
-                    # TODO: Make these times relative (e.g. started 1 hour ago)
-                    message = message + f"   Scheduled to start at: {outage['startdate']}{NL}"
-                    message = message + (
-                        f"   Scheduled to end at: "
-                        f"{outage['enddate'] if outage['enddate'] else 'Unknown'}{NL}"
+                    message += f" * {format_title(outage)}{NL}"
+
+                    start_time = self.date_validation(outage.get("startdate"))
+                    relative_start_time = self.get_relative_time(start_time - now)
+
+                    message += (
+                        f"   Scheduled to start :" + f"{start_time}" + f"{relative_start_time}{NL}"
                     )
+                    message += (
+                        f"   Estimated to end: "
+                        f"{outage.get('enddate') if outage.get('enddate') else 'Unknown'}{NL}"
+                        f"{self.get_relative_time(outage.get('enddate') - now  if outage.get('enddate') else datetime.datetime.max - now)}{NL}"
+                    )
+
+            await evt.respond(message)
+
+    def date_validation(self, date_string):
```

Comment:
I think the name of the function is a little confusing since it's formatting the date rather that just validating it _is_ a date, so I would suggest renaming it to `format_date(self, date_time)` or something like that.

Reply:
Okay, okay that makes sense. Will work on my naming conventions